corehq/util/string_utils.py

```diff
@@ -12,5 +12,9 @@ def ensure_unicode(s):
     if isinstance(s, six.text_type):
         return s
     elif isinstance(s, six.binary_type):
-        return s.decode('utf-8')
+        try:
+            return s.decode('utf-8')
+        except UnicodeDecodeError:
+            # https://sentry.io/dimagi/commcarehq/issues/391378081/
+            return s.decode('latin1')
```

Comment:
Hmm, guessing at the encoding is not ideal.

A better fix for that bug would be to have [`get_username_and_password_from_request`](https://github.com/dimagi/commcare-hq/blob/c888c470f18661f8c5fede684c7545619a17cd2e/corehq/apps/domain/auth.py#L76) return correctly decoded strings as specified by the standard. Unfortunately, there is no standard character encoding due to the way the HTTP Basic Authentication standard was originally defined. [RFC-7617](https://tools.ietf.org/html/rfc7617#section-2.1) outlines a way for servers to request clients to use UTF-8, but there is no guarantee that this will be followed.

In any case, I think it would be better move this try/except guessing thing into `get_username_and_password_from_request` rather than putting it here in a general utility function that will get used all over the place.